src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,96 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int HOURS_INDEX = 2;
+    private static final int INCOME_INDEX = 3;
+    private static final int NAME_INDEX = 1;
+    private static final String DATA_DELIMITER = " ";
+    private static final String INDEX_NAME_DELIMITER = ":";
+    private static final String INDEX_NAME_PAIRS_DELIMITER = "|";
+    private static final String REPORT_DELIMITER = " - ";
+    private static final String REPORT_HEADER = "Report for period ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String indexedNameString = buildIndexedNameString(names);
+        int[] salaries = new int[names.length];
+        for (String dataRecord : data) {
+            String[] records = dataRecord.split(DATA_DELIMITER);
+            if (!indexedNameString.contains(records[NAME_INDEX])) {
+                continue;
+            }
+            LocalDate parsedRecordDate = parseStringToLocalDate(records[DATE_INDEX]);
+            if (parsedRecordDate.isBefore(parseStringToLocalDate(dateFrom))) {
+                continue;
+            }
+            if (parsedRecordDate.isAfter(parseStringToLocalDate(dateTo))) {
```

Comment:
Similar to the previous comment, 'dateTo' is also parsed inside the loop. It's more efficient to parse it once outside the loop and use the parsed value for comparison inside the loop.